### ISSUE TITLE
SystemUiOverlayStyle#systemNavigationBarDividerColor doesn't work.

### ISSUE THREAD

**Issue by colinrtwhite** [flutter/flutter#26478]
## Steps to Reproduce

1. Set the `systemNavigationBarDividerColor` using `SystemChrome.setSystemUIOverlayStyle`.
2. Run the app on an Android device running Android P or above.
3. The navigation bar divider colour is not set.

Here's an example project: [nav_bar_bug.zip](https://github.com/flutter/flutter/files/2752590/nav_bar_bug.zip)

## Logs

```
No issues found! (ran in 1.8s)
```

```
[✓] Flutter (Channel dev, v1.1.9, on Mac OS X 10.14.2 18C54, locale en-CA)
    • Flutter version 1.1.9 at /Users/Colin/flutter
    • Framework revision 1407091bfb (4 days ago), 2019-01-08 20:40:19 -0800
    • Engine revision e5ec3cf3ea
    • Dart version 2.1.1 (build 2.1.1-dev.0.1 2cb346bd0c)

[✓] Android toolchain - develop for Android devices (Android SDK version 28.0.3)
    • Android SDK at /Users/Colin/Library/Android/sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-28, build-tools 28.0.3
    • ANDROID_HOME = /Users/Colin/Library/Android/sdk
    • Java binary at: /Applications/Android Studio 3.3 Preview.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1248-b01)
    • All Android licenses accepted.

[✓] iOS toolchain - develop for iOS devices (Xcode 10.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 10.1, Build version 10B61
    • ios-deploy 1.9.4
    • CocoaPods version 1.5.3

[!] Android Studio (version 3.3)
    • Android Studio at /Applications/Android Studio 3.3 Preview.app/Contents
    ✗ Flutter plugin not installed; this adds Flutter specific functionality.
    ✗ Dart plugin not installed; this adds Dart specific functionality.
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1248-b01)

[!] Android Studio (version 3.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    ✗ Flutter plugin not installed; this adds Flutter specific functionality.
    ✗ Dart plugin not installed; this adds Dart specific functionality.
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1136-b06)

[✓] IntelliJ IDEA Community Edition (version 2018.3.2)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin version 31.3.4
    • Dart plugin version 183.4886.3

[!] VS Code (version 1.30.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    ✗ Flutter extension not installed; install from
      https://marketplace.visualstudio.com/items?itemName=Dart-Code.flutter

[✓] Connected device (1 available)
    • Pixel 3 • 8BDX18F5Y • android-arm64 • Android 9 (API 28)

! Doctor found issues in 3 categories
```

**Comment by zoechi** [comment]
To investigate we would need a minimal runnable reproduction
as a single file so that we can just copy your code into `lib/main.dart` of a new project and run to reproduce.

**Comment by colinrtwhite** [comment]
Sure, calling this function in `build` method of main widget should do the trick:

```
  static void initSystemUIOverlayStyle() async {
    if (!Platform.isAndroid) {
      return;
    }

    SystemChrome.setSystemUIOverlayStyle(SystemUiOverlayStyle.dark.copyWith(
      systemNavigationBarColor: Colors.white,
      systemNavigationBarIconBrightness: Brightness.dark,
      systemNavigationBarDividerColor: Colors.black
    ));
  }
```

**Comment by zoechi** [comment]
Using `AnnotatedRegion` might also work. I haven't tried it myself yet https://github.com/flutter/flutter/pull/24989/files

**Comment by iapicca** [comment]
Hi @colinrtwhite 
testing the code you provided with the latest master

<details><summary>code</summary>

```dart
import 'dart:io';
import 'package:flutter/material.dart';
import 'package:flutter/services.dart';

void main() => runApp(MyApp());

class MyApp extends StatelessWidget {
  // This widget is the root of your application.
  @override
  Widget build(BuildContext context) {
    initSystemUIOverlayStyle();

    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        // This is the theme of your application.
        //
        // Try running your application with "flutter run". You'll see the
        // application has a blue toolbar. Then, without quitting the app, try
        // changing the primarySwatch below to Colors.green and then invoke
        // "hot reload" (press "r" in the console where you ran "flutter run",
        // or simply save your changes to "hot reload" in a Flutter IDE).
        // Notice that the counter didn't reset back to zero; the application
        // is not restarted.
        primarySwatch: Colors.blue,
      ),
      home: MyHomePage(title: 'Flutter Demo Home Page'),
    );
  }

  static void initSystemUIOverlayStyle() async {
    if (!Platform.isAndroid) {
      return;
    }

    SystemChrome.setSystemUIOverlayStyle(SystemUiOverlayStyle.dark.copyWith(
      systemNavigationBarColor: Colors.white,
      systemNavigationBarIconBrightness: Brightness.dark,
      systemNavigationBarDividerColor: Colors.black
    ));
  }
}

class MyHomePage extends StatefulWidget {
  MyHomePage({Key key, this.title}) : super(key: key);

  // This widget is the home page of your application. It is stateful, meaning
  // that it has a State object (defined below) that contains fields that affect
  // how it looks.

  // This class is the configuration for the state. It holds the values (in this
  // case the title) provided by the parent (in this case the App widget) and
  // used by the build method of the State. Fields in a Widget subclass are
  // always marked "final".

  final String title;

  @override
  _MyHomePageState createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  int _counter = 0;

  void _incrementCounter() {
    setState(() {
      // This call to setState tells the Flutter framework that something has
      // changed in this State, which causes it to rerun the build method below
      // so that the display can reflect the updated values. If we changed
      // _counter without calling setState(), then the build method would not be
      // called again, and so nothing would appear to happen.
      _counter++;
    });
  }

  @override
  Widget build(BuildContext context) {
    // This method is rerun every time setState is called, for instance as done
    // by the _incrementCounter method above.
    //
    // The Flutter framework has been optimized to make rerunning build methods
    // fast, so that you can just rebuild anything that needs updating rather
    // than having to individually change instances of widgets.
    return Scaffold(
      appBar: AppBar(
        // Here we take the value from the MyHomePage object that was created by
        // the App.build method, and use it to set our appbar title.
        title: Text(widget.title),
      ),
      body: Center(
        // Center is a layout widget. It takes a single child and positions it
        // in the middle of the parent.
        child: Column(
          // Column is also layout widget. It takes a list of children and
          // arranges them vertically. By default, it sizes itself to fit its
          // children horizontally, and tries to be as tall as its parent.
          //
          // Invoke "debug painting" (press "p" in the console, choose the
          // "Toggle Debug Paint" action from the Flutter Inspector in Android
          // Studio, or the "Toggle Debug Paint" command in Visual Studio Code)
          // to see the wireframe for each widget.
          //
          // Column has various properties to control how it sizes itself and
          // how it positions its children. Here we use mainAxisAlignment to
          // center the children vertically; the main axis here is the vertical
          // axis because Columns are vertical (the cross axis would be
          // horizontal).
          mainAxisAlignment: MainAxisAlignment.center,
          children: <Widget>[
            Text(
              'You have pushed the button this many times:',
            ),
            Text(
              '$_counter',
              style: Theme.of(context).textTheme.display1,
            ),
          ],
        ),
      ),
      floatingActionButton: FloatingActionButton(
        onPressed: _incrementCounter,
        tooltip: 'Increment',
        child: Icon(Icons.add),
      ), // This trailing comma makes auto-formatting nicer for build methods.
    );
  }
}

```

</details>

<details><summary>screenshots</summary>

@ launch

![Screenshot_20200729-111433](https://user-images.githubusercontent.com/45637986/88775203-161fd400-d18d-11ea-87e6-0e01c87f07dc.jpg)



after minimising and back
![Screenshot_20200729-111425](https://user-images.githubusercontent.com/45637986/88775095-f8526f00-d18c-11ea-86c3-85f2cd306c7e.jpg)



</details>

<details><summary>doctor</summary>

```console
[✓] Flutter (Channel master, 1.21.0-6.0.pre.90, on Mac OS X 10.15.5 19F101, locale en-GB)
    • Flutter version 1.21.0-6.0.pre.90 at /Users/nevercode/development/flutter_master
    • Framework revision 531ee9452a (2 hours ago), 2020-07-29 02:26:03 -0400
    • Engine revision 1ba310017c
    • Dart version 2.10.0 (build 2.10.0-1.0.dev 24c7666def)


[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
    • Android SDK at /Users/nevercode/Library/Android/sdk
    • Platform android-29, build-tools 29.0.2
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 11.5)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 11.5, Build version 11E608c
    • CocoaPods version 1.9.0

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.0)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 46.0.2
    • Dart plugin version 193.7361
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)

[✓] VS Code
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.12.2

[✓] Connected device (4 available)
    • Pixel 3a (mobile) • 965AY0WP5C • android-arm64  • Android 10 (API 29)
    • macOS (desktop)   • macos      • darwin-x64     • Mac OS X 10.15.5 19F101
    • Web Server (web)  • web-server • web-javascript • Flutter Tools
    • Chrome (web)      • chrome     • web-javascript • Google Chrome 84.0.4147.105

• No issues found!
```

</details>

it seems to apply the seeings only after minimising and restoring the app
Thank you

**Comment by jacek-marchwicki** [comment]
@colinrtwhite Looks like it is possible to make the navigation bar color change work. This is an example: https://github.com/jacek-marchwicki/flutter-insets-bug2/commit/74e17edc48c4045e29cfee6524e4f86f979be9f0.

There is a small issue: https://github.com/flutter/flutter/issues/66429 but if you don't need to change the color while content scrolls it shouldn't be a problem for you.

**Comment by brightseagit** [comment]
Hi, @zoechi I think you misunderstood the OP. The OP wants to say that using systemNavigationBarDividerColor: Colors.black doesn't do anything. And I have also faced the same issue.
Everything else in SystemUiOverlayStyle worked fine except systemNavigationBarDividerColor: . I'm pasting below the code I used.

```
AnnotatedRegion<SystemUiOverlayStyle>(
      value: SystemUiOverlayStyle(
        systemNavigationBarDividerColor: Colors
            .black,
        systemNavigationBarColor: Colors.white,
        statusBarColor: Colors.white,
        systemNavigationBarIconBrightness: Brightness.dark,
        statusBarIconBrightness: Brightness.dark,
      ),
```

This is how it looks in my app on a device running Android R/11.
![image](https://user-images.githubusercontent.com/72334435/107661103-f4acb280-6cae-11eb-979b-3a7784796546.png)
As you can see that there is no divider between systemNavigationBar and bottomNavigationBar.

Please let me know if I'm missing or misunderstanding anything here. Thank you.

**Comment by brightseagit** [comment]
@TahaTesser I think this issue has been fixed in flutter stable 2.0.1 
As expected the  systemNavigationBarDividerColor is visible now :-
![image](https://user-images.githubusercontent.com/72334435/110282804-20e4f600-8005-11eb-8824-f033594b6f46.png)

**Comment by TahaTesser** [comment]
Hi @colinrtwhite 
Thanks for the ping, it does work on the latest `stable` channel now

![2021_03_09_18_06_59_2](https://user-images.githubusercontent.com/48603081/110471671-6804e100-8102-11eb-8975-c3ec52277ba0.jpg)

<details> 
<summary>flutter doctor -v</summary> 

```bash
[✓] Flutter (Channel stable, 2.0.1, on macOS 11.2.2 20D80 darwin-x64, locale en-GB)
    • Flutter version 2.0.1 at /Users/tahatesser/Code/flutter_stable
    • Framework revision c5a4b4029c (5 days ago), 2021-03-04 09:47:48 -0800
    • Engine revision 40441def69
    • Dart version 2.12.0

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at /Volumes/Extreme/SDK
    • Platform android-30, build-tools 30.0.3
    • ANDROID_HOME = /Volumes/Extreme/SDK
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Volumes/Extreme/Xcode.app/Contents/Developer
    • Xcode 12.4, Build version 12D4e
    • CocoaPods version 1.10.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)

[✓] VS Code (version 1.54.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.20.0

[✓] Connected device (4 available)
    • SM M025F (mobile)    • R9ZR205XX0A               • android-arm    • Android 10 (API 29)
    • Taha’s iPad (mobile) • 00008020-000255113EE8402E • ios            • iOS 14.4
    • macOS (desktop)      • macos                     • darwin-x64     • macOS 11.2.2 20D80 darwin-x64
    • Chrome (web)         • chrome                    • web-javascript • Google Chrome 89.0.4389.82

• No issues found!
```

</details>
